rslib/src/scheduler/fsrs/retention.rs

```diff
@@ -135,15 +135,22 @@ impl Collection {
                     r.review_kind == RevlogReviewKind::Review
                         && r.button_chosen > 0
                         && r.taken_millis > 0
+                        && r.taken_millis < 1200000 // 20 minutes
                 })
                 .sorted_by(|a, b| a.button_chosen.cmp(&b.button_chosen))
                 .group_by(|r| r.button_chosen)
                 .into_iter()
                 .for_each(|(button_chosen, group)| {
-                    let group_vec = group.into_iter().map(|r| r.taken_millis).collect_vec();
-                    let average_secs =
-                        group_vec.iter().sum::<u32>() as f64 / group_vec.len() as f64 / 1000.0;
-                    arr[button_chosen as usize - 1] = average_secs
+                    let mut group_vec = group.into_iter().map(|r| r.taken_millis).collect_vec();
+                    group_vec.sort_unstable();
+                    let median_millis = if group_vec.len() % 2 == 0 {
+                        let mid = group_vec.len() / 2;
+                        (group_vec[mid - 1] + group_vec[mid]) as f64 / 2.0
+                    } else {
+                        group_vec[group_vec.len() / 2] as f64
+                    };
+                    let median_secs = median_millis / 1000.0;
+                    arr[button_chosen as usize - 1] = median_secs;
```

Comment:
The median calculation is repeated 3 times - it would be nicer if this was factored out into a separate function, and that would also make it easy to add a simple unit test that makes sure it handles 0, 1 and 2 element vecs correctly.

Reply:
I have refactored the code just now.

Reply:
Thanks!